### ISSUE TITLE
Kotlin support [feature request]

### ISSUE THREAD

**Issue by neworld** [droidchef/hops#4]
Nowadays more and more Android & Java projects use [Kotlin](https://kotlinlang.org/). I suppose full Kotlin support is not easy and requires parsing all files, but would be nice if `*.kt` navigates to other `*.kt` files with same rules as in Java.

**Comment by ubuntudroid** [comment]
@DVDAndroid would you mind creating a PR to get this merged into the official Chrome extension? Or is it still WIP and not yet ready to be merged? Would love to try this out in my own Kotlin projects. :)

**Comment by DVDAndroid** [comment]
@ubuntudroid Yeah, I'll create a PR. Could you test it before my PR? ~[download](https://github.com/droidchef/hops/files/xx1455/ext.zip)~

**Comment by neworld** [comment]
I checked and seems doesn't work. It could be easily checked at my repository [in this file](https://github.com/neworld/ClimbTimer/blob/26eb5e0ed272b6dcc88119c7261693177bf6861c/app/src/main/java/lt/neworld/climbtimer/Main.kt)

That `TimerController` in the [Main.kt:14](https://github.com/neworld/ClimbTimer/blob/26eb5e0ed272b6dcc88119c7261693177bf6861c/app/src/main/java/lt/neworld/climbtimer/Main.kt#L14) could be linked to [TimerController.kt](https://github.com/neworld/ClimbTimer/blob/26eb5e0ed272b6dcc88119c7261693177bf6861c/app/src/main/java/lt/neworld/climbtimer/controllers/TimerController.kt)

The path from package could be decoded in the same manner as in java. No difference in this case.

**Comment by DVDAndroid** [comment]
@neworld Something is not working at line 58 of [content.js](https://github.com/DVDAndroid/hops/blob/kotlin-support/content.js#L58). The CSS selector seems that it doesn't find any `td` which contains `TimerController`

**Comment by neworld** [comment]
If I understand correctly, github formats kotlin code a bit different. In Java all class names are wrapped in span, but in kotlin isn't
```
<td id="LC14" class="blob-code blob-code-inner js-file-line" style="...">        
  primaryStage
  <span class="pl-k">.</span>
  scene 
  <span class="pl-k">=</span>
  TimerController
  <span class="pl-k">.</span>
  newScene()
</td>
```

**Comment by DVDAndroid** [comment]
Hi @neworld @ubuntudroid, I just updated my fork and the extension. Can you try it before the PR? 
~[hops.zip](https://github.com/droidchef/hops/files/xx3703/hops.zip)~

Here's my last commit that workarounds last issue: https://github.com/DVDAndroid/hops/commit/d0a3ab6127d4bd39e8889136619e0e8481824f2f

**Comment by neworld** [comment]
It works. Nice work!

**Comment by droidchef** [comment]
Looks like a good start!

It fails to link classes correctly in this repo for me 
https://github.com/TonicArtos/SuperSLiM/blob/master/library/src/main/kotlin/com/tonicartos/superslim/layout.kt

**Comment by neworld** [comment]
As expected. This plugin works nicely with java, because java has strict rules between packages and file/dir names. In the kotlin is easy to break them

**Comment by DVDAndroid** [comment]
@droidchef Fixed CSS highlighting, but it doesn't find the file. ~[hops.zip](https://github.com/droidchef/hops/files/xx4156/hops.zip)~

**Comment by neworld** [comment]
There is another issue. Sometimes detects partial of the class. [In this file](https://github.com/neworld/ClimbTimer/blob/7ba629a09bfe57a9c5c020c8bf79af2501e3b148/app/src/main/java/lt/neworld/climbtimer/controllers/TimerController.kt#L24) plugin wraps only `Timer` and links to it, instead of `TimerController`

**Comment by DVDAndroid** [comment]
@neworld Try now please:  ~[hops.zip](https://github.com/droidchef/hops/files/xx4235/hops.zip)~

**Comment by neworld** [comment]
Nop, still the same.

**Comment by DVDAndroid** [comment]
Fixed some small bugs. Looks good to me. Tested also on some Android files written in Kotlin. [hops.zip](https://github.com/droidchef/hops/files/993674/hops.zip). Remove old versions of the extension before install

**Comment by neworld** [comment]
Much better. I can't find any more issues

**Comment by droidchef** [comment]
@DVDAndroid I do see the fixed CSS thingy. Great job with that. Can you send a PR for it? So I can close the issue related to that. 

The Kotlin thing doesn't work perfectly still from what I can check. 

See this repo
https://github.com/TonicArtos/SuperSLiM/blob/master/library/src/main/kotlin/com/tonicartos/superslim/layout.kt

I don't know what is so weird about this :(

**Comment by DVDAndroid** [comment]
I've opened the PR. The problem of Kotlin is that you can create multiple classes in `kt` files and you can call the `kt` file in different way.
I think you can close issue #9 because the issue doesn't exist with current version of the extension (available from Chrome Web Store) and it is fixed in one of my commits.

**Comment by neworld** [comment]
For full Kotlin support, the plugin needs to scan all files. However, if there is any chance to scan all files, it would be possible skip these classes.

**Comment by DVDAndroid** [comment]
Or I think it's possible to make a HTTP request to the class, if it returns 404, don't highlight on the file...
